src/osd/PrimaryLogPG.cc

```diff
@@ -641,6 +641,9 @@ void PrimaryLogPG::block_for_clean(
 {
   dout(20) << __func__ << ": blocking object " << oid
 	   << " on primary repair" << dendl;
+ if (objects_blocked_clean_to_primary_repair.count(oid) == 0) {
+    objects_blocked_clean_to_primary_repair.insert(oid);
```

Comment:
For an STL set shouldn't just insert() be fine. I would imagine it is just as efficient for the insert to notice the object is already there.